social-ui/src/ForYouCard.js

```diff
@@ -0,0 +1,35 @@
+import React, { Component } from "react";
+import Meta from "./Meta";
+import yourArticles from "./_data/your-articles.json";
+
+class CardFor extends Component {
+  constructor() {
+    super();
+    this.state = {
+      yourArticles
+    };
+  }
+
+  render() {
+    return this.state.yourArticles.map((article, idx) => (
+      <div className="card" key={idx}>
+        <div className="image-holder">
+          <img src={article.image} alt={article.title} />
+        </div>
+        <h3>{article.title}</h3>
+        <p>{article.description}</p>
+        <div className="metadata">
+        <span>
+          <Meta article={article} key={idx} />
+        </span>
+        <span>{/*article.postedDate.toDateString()   Why won't this work? What's the point of writing HTML in javascript if I can't use javascript?*/}</span>
```

Comment:
So JSX inside a return can only do basic operations (inline ifs mostly) generally you want to do all the formatting and map functions outside of the return statement, and have only the HTML/JSX inside the render.

Reply:
Correction, so what I wrote above is not really correct, although you do want to try and minimize logic inside the JSX render function, you can get your code to work if you do :
`<span>{new Date(article.postedDate.toDateString()}</span>`
it did not originally work because you were calling toDateString() on a string when it is a Date object function.